<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageCalibrator</class>
 <widget class="QDialog" name="ImageCalibrator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>626</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>463</width>
    <height>485</height>
   </size>
  </property>
  <property name="focusPolicy">
   <enum>Qt::ClickFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Sprite editor</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../_resources/images.qrc">
    <normaloff>:/images/Icon16.png</normaloff>:/images/Icon16.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_3" columnstretch="0,0,0,1000">
   <item row="0" column="3">
    <widget class="FrameTuneScene" name="preview">
     <property name="frameShape">
      <enum>QFrame::WinPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
    </widget>
   </item>
   <item row="0" column="2">
    <widget class="QFrame" name="toolBox">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QToolButton" name="toolDrag">
        <property name="toolTip">
         <string>Drag the frame through entire sprite</string>
        </property>
        <property name="text">
         <string>Drag mode</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_move.png</normaloff>:/edit/edit_move.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolSelect">
        <property name="toolTip">
         <string>Select fragment and cut it, move it into necessary place and double-click to paste</string>
        </property>
        <property name="text">
         <string>Select and move</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_select.png</normaloff>:/edit/edit_select.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolSelectCopy">
        <property name="toolTip">
         <string>Select fragment and copy it, move it into necessary place and double-click to paste</string>
        </property>
        <property name="text">
         <string>Select and copy</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_select_copy.png</normaloff>:/edit/edit_select_copy.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolPencil">
        <property name="toolTip">
         <string>Pencil to draw</string>
        </property>
        <property name="text">
         <string>Pencil</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_pencil.png</normaloff>:/edit/edit_pencil.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolPicker">
        <property name="toolTip">
         <string>Pick a color</string>
        </property>
        <property name="text">
         <string>Color picker</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_pick.png</normaloff>:/edit/edit_pick.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolRubber">
        <property name="toolTip">
         <string>Rubber</string>
        </property>
        <property name="text">
         <string>Rubber</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_rubber.png</normaloff>:/edit/edit_rubber.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolLine">
        <property name="toolTip">
         <string>Draw a line</string>
        </property>
        <property name="text">
         <string>Draw lines</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_line.png</normaloff>:/edit/edit_line.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolRects">
        <property name="toolTip">
         <string>Draw a rectangle</string>
        </property>
        <property name="text">
         <string>Draw rectangles</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_rect.png</normaloff>:/edit/edit_rect.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolQuickActions">
        <property name="toolTip">
         <string>Perform one of actions on a current frame or the whole sprite</string>
        </property>
        <property name="text">
         <string>Select and move</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/quick_actions.png</normaloff>:/edit/quick_actions.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="popupMode">
         <enum>QToolButton::InstantPopup</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolUndo">
        <property name="toolTip">
         <string>Undo a recent change</string>
        </property>
        <property name="text">
         <string>Undo a recent change</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_undo.png</normaloff>:/edit/edit_undo.png</iconset>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolRedo">
        <property name="toolTip">
         <string>Redo a recent change</string>
        </property>
        <property name="text">
         <string>Redo a recent change</string>
        </property>
        <property name="icon">
         <iconset resource="../_resources/images.qrc">
          <normaloff>:/edit/edit_redo.png</normaloff>:/edit/edit_redo.png</iconset>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>479</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="7" column="1">
       <widget class="QSpinBox" name="CropW">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="17" column="0" colspan="2">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QLabel" name="label_8">
        <property name="font">
         <font>
          <pointsize>11</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Drag frame offset:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Drag offset X</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Drag offset Y</string>
        </property>
       </widget>
      </item>
      <item row="15" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Opacity</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0" colspan="2">
       <widget class="QPushButton" name="Reset">
        <property name="text">
         <string>Reset calibration</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Frame X</string>
        </property>
       </widget>
      </item>
      <item row="14" column="0" colspan="2">
       <widget class="QLabel" name="label_2">
        <property name="font">
         <font>
          <pointsize>11</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Referrence frame:</string>
        </property>
       </widget>
      </item>
      <item row="15" column="1">
       <widget class="QSlider" name="refOpacity">
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="13" column="0" colspan="2">
       <widget class="QPushButton" name="openFrameInEditor">
        <property name="text">
         <string>Open frame in editor...</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="FrameY">
        <property name="maximum">
         <number>9</number>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Crop right</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Crop bottom</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QSpinBox" name="OffsetX">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>-100</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <widget class="QCheckBox" name="hitboxMove">
        <property name="toolTip">
         <string>When using mouse to move the picture, also move hitbox too. Use this if you want to align the frame without breaking the calibration setup.</string>
        </property>
        <property name="text">
         <string>Hitbox move</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Frame Y</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="QCheckBox" name="moveAllFrames">
        <property name="toolTip">
         <string>Move all frames of the sprite synchroniously with current. Use this to simplify the general re-aligning work.</string>
        </property>
        <property name="text">
         <string>Move all frames</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label_7">
        <property name="font">
         <font>
          <pointsize>11</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Current frame (10x10):</string>
        </property>
       </widget>
      </item>
      <item row="16" column="0" colspan="2">
       <widget class="QPushButton" name="refClear">
        <property name="text">
         <string comment="reference frame">Clear</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QSpinBox" name="OffsetY">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>-100</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QSpinBox" name="CropH">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="FrameX">
        <property name="maximum">
         <number>9</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QPushButton" name="Matrix">
        <property name="text">
         <string>Show sprite map</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="11" column="0" colspan="2">
       <widget class="QCheckBox" name="dublPixDrawMode">
        <property name="toolTip">
         <string>Forces any draw operations to be aligned by 2x2 pixel cells.</string>
        </property>
        <property name="text">
         <string>2-pix draw mode</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="4">
    <widget class="QGroupBox" name="groupBox">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>60</height>
      </size>
     </property>
     <property name="title">
      <string>Save image</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QPushButton" name="WritePNG_GIF">
        <property name="text">
         <string>Write PNG+GIFs</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="WritePNG">
        <property name="text">
         <string>Write PNG</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="WriteGIF">
        <property name="text">
         <string>Write masked GIFs</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FrameTuneScene</class>
   <extends>QFrame</extends>
   <header>main/frame_tune_scene.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../_resources/images.qrc"/>
 </resources>
 <connections/>
</ui>
